packages/itwin/tree-widget/src/tree-widget-react/components/trees/common/TreeWidgetViewport.ts

```diff
@@ -0,0 +1,247 @@
+   *
+   * This set takes precedence over category and sub-category display and per-model category override.
+   *
+   * **NOTE** This set should take precedence over the [[alwaysDrawn]] set - if an element is present in both sets, it is never drawn.
+   */
+  neverDrawn: ReadonlySet<Id64String> | undefined;
+  /**
+   * Ids of elements which should not be displayed in the viewport, regardless of category and sub-category visibility.
+   *
+   * @param elementIds The Ids of the elements that should not be displayed.
+   */
+  setNeverDrawn: (elementIds: Set<Id64String>) => void;
+  /** Should clear the set of never-drawn elements. */
+  clearNeverDrawn: () => void;
+  /**
+   * Ids of elements which should be displayed.
+   *
+   * Elements in this set should be displayed only when model is displayed as well.
+   *
+   * This set takes precedence over category and sub-category display and per-model category override.
+   *
+   * **NOTE** The [[neverDrawn]] set takes precedence - if an element is present in both sets, it is never drawn.
+   */
+  alwaysDrawn: ReadonlySet<Id64String> | undefined;
+  /**
+   * Ids of elements which should be displayed in the viewport, regardless of category and sub-category visibility.
+   *
+   * These elements should be displayed only when their models are displayed as well.
+   * @param elementIds The Ids of the elements that should be displayed.
+   * @param exclusive If true, *only* the specified elements should be drawn. When this is set to true `isAlwaysDrawnExclusive` should return true, otherwise `isAlwaysDrawnExclusive` should return false.
+   */
+  setAlwaysDrawn: (props: { elementIds: Set<Id64String>; exclusive?: boolean }) => void;
+  /** Should clear the set of always-drawn elements. */
+  clearAlwaysDrawn: () => void;
+  /**
+   * Should return true if elements in the [[alwaysDrawn]] set are the *only* elements that are displayed when their models are visible.
+   *
+   * Should be set to true when `setAlwaysDrawn` is called with `exclusive` flag.
+   *
+   * Should be set to false when `setAlwaysDrawn` is called with `exclusive` set to false/undefined or when `clearAlwaysDrawn` is called.
+   */
+  readonly isAlwaysDrawnExclusive: boolean;
+  /** Event that should be raised when set of always-drawn elements changes. */
+  onAlwaysDrawnChanged: BeEvent<() => void>;
+  /** Event that should be raised when set of never-drawn elements changes. */
+  onNeverDrawnChanged: BeEvent<() => void>;
+  /** Event that should be raised when per-model category overrides changes. */
+  onPerModelCategoriesOverridesChanged: BeEvent<() => void>;
+  /** Event that should be raised when category display changes. */
+  onDisplayedCategoriesChanged: BeEvent<() => void>;
+  /** Event that should be raised when model display changes. */
+  onDisplayedModelsChanged: BeEvent<() => void>;
+  /**
+   * Event that should be raised when model, category or sub-category display changes.
```

Comment:
is this really true, meaning that changing model display fires `onDisplayedModelsChanged` and this one too?